test/integration/apiserver/crd_validation_expressions_test.go

```diff
@@ -523,8 +523,8 @@ func TestCustomResourceValidatorsWithBlockingErrors(t *testing.T) {
 			if err == nil {
 				t.Fatal("Expected create of invalid custom resource to fail")
 			} else {
-				if !strings.Contains(err.Error(), "failed rule: self.spec.x + self.spec.y") {
-					t.Fatalf("Expected error to contain %s but got %v", "failed rule: self.spec.x + self.spec.y", err.Error())
+				if !strings.Contains(err.Error(), "self.spec.x + self.spec.y must be greater than or equal to") {
+					t.Fatalf("Expected error to contain %s but got %v", "self.spec.x + self.spec.y must be greater than or equal to", err.Error())
```

Comment:
Have the full string here for testing? For test case, we should specify what we expected to have from `format([(has(self.status) ? self.status.z : 0)]`

Reply:
Right, I've added the output from the `%d` portion of the `string.format` call to make sure that part works as well too.